#165146389 Implement a landing page for the application

olajide1234 opened the pull request:
#### What does this PR do?
Implements the landing page using react and redux

#### Description of Task to be completed?
Users should be welcomed with a landing page when they visit the application's URL

#### How should this be manually tested?
- Visit the app URL `https://vidar-ah-frontend-staging.herokuapp.com/`

#### Any background context you want to provide?
- N/A

#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/165146389

#### Screenshots (if appropriate)

#### Questions:
N/A

kleva-j commented on src/redux/actions/actionTypes.js:
```diff
@@ -1,2 +1,15 @@
+export const CREATE_ARTICLE = 'CREATE_ARTICLE';
+export const SET_ARTICLE = 'SET_ARTICLE';
+export const SET_ARTICLE_ERROR = 'SET_ARTICLE_ERROR';
+export const SIGNUP = 'SIGNUP';
```
Please Include only the action types you need.

olajide1234 replied:
Updated. Thanks.

olajide1234 committed:
feat(landing): implement landing page

- create all components
- collate components into view
- hook component to backend

[Delivers #165146389]

9jaswag commented on src/__tests__/components/banner.spec.js:
```diff
@@ -0,0 +1,17 @@
+import { BrowserRouter } from 'react-router-dom';
+import Banner from '../../components/banner/Index';
+
+describe('Test banner mounts', () => {
```
@olajide1234 does "Test banner mounts" describe what we're testing?

9jaswag commented on src/__tests__/components/contentHeader.spec.js:
```diff
@@ -0,0 +1,14 @@
+
+  it('renders correctly', () => {
+    component = mount(
+      <ContentHeader textHeader="Test header" />
```
@olajide1234 same here. Refer to @Dubby20 on how to name describe blocks and share it with the team.

9jaswag commented on src/__tests__/components/headerText.spec.js:
```diff
@@ -0,0 +1,14 @@
+import { mount } from 'enzyme';
+import HeaderText from '../../components/headerText/Index';
+
+describe('Test header text', () => {
```
@olajide1234 fix this here and any other place it occurs

9jaswag commented on src/assets/default.scss:
```diff
@@ -1,25 +1,133 @@
+}
+
+.signup-form {
+  width: 80%;
```
depending on the use case, I feel that Bootstrap can solve for this, instead of setting a width.

9jaswag commented on src/components/articleSummary/Index.jsx:
```diff
@@ -0,0 +1,57 @@
+  const {
+    src, header, name, time
+  } = props;
+  const date = new Date(time).getTime();
```
@olajide1234 I noticed this in Jacinta's PR. If this is a logic that will be repeated throughout the app, we should extract it into a function.

9jaswag commented on src/components/rectArticleSummary/Index.jsx:
```diff
@@ -0,0 +1,47 @@
+  const { header, name, time } = props;
+  const date = new Date(time).getTime();
+  const now = new Date().getTime();
+  const hrs = Math.floor((now - date) / 1000 / 60 / 60);
```
@olajide1234 we should definitely extract this and use it to replace everywhere it's being used in the app

9jaswag commented on src/redux/actions/getArticles.js:
```diff
@@ -0,0 +1,29 @@
+
+export const getArticles = params => async (dispatch) => {
+  try {
+    const { data } = await axios.get(`https://vidar-ah-backend-production.herokuapp.com/api/articles/order?type=latest&amount=${params}`);
```
@olajide1234 you can have your URL in a variable...so this looks cleaner

9jaswag commented on src/redux/actions/getArticles.js:
```diff
@@ -0,0 +1,29 @@
+
+export const getTrendingArticles = params => async (dispatch) => {
+  try {
+    const { data } = await axios.get(`https://vidar-ah-backend-production.herokuapp.com/api/articles/order?type=latest&amount=${params}`);
```
@olajide1234 same here

9jaswag reviewed (changes requested):
Attend to these comments and re-add me with the proper labels

9jaswag commented:
@olajide1234 why is this not ready for me to review?

9jaswag commented on src/views/landingPage/Index.jsx:
```diff
@@ -0,0 +1,156 @@
+    props.getTrendingArticles(6).then();
+  }, {});
+
+  // const ArticleSummaryStack = () => props.trendingArticles.map(data => (
```
@olajide1234 why do we have commented out code here?

olajide1234 replied:
Removed now.
